webcrypto/subtle_crypto.go

```diff
@@ -496,46 +496,66 @@ func (sc *SubtleCrypto) Digest(algorithm goja.Value, data goja.Value) *goja.Prom
 //
 // The `keyUsages` parameter is an array of strings indicating what the key can be used for.
 func (sc *SubtleCrypto) GenerateKey(algorithm goja.Value, extractable bool, keyUsages []CryptoKeyUsage) *goja.Promise {
-	promise, resolve, reject := promises.New(sc.vu)
+	rt := sc.vu.Runtime()
 
-	normalized, err := normalizeAlgorithm(sc.vu.Runtime(), algorithm, OperationIdentifierGenerateKey)
-	if err != nil {
-		reject(err)
-		return promise
-	}
+	var keyGenerator KeyGenerator
+
+	err := func() error {
+		normalized, err := normalizeAlgorithm(rt, algorithm, OperationIdentifierGenerateKey)
+		if err != nil {
+			return err
+		}
 
-	keyGenerator, err := newKeyGenerator(sc.vu.Runtime(), normalized, algorithm)
+		keyGenerator, err = newKeyGenerator(rt, normalized, algorithm)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}()
+
+	promise, resolve, reject := rt.NewPromise()
```

Comment:
If I'm not wrong, with the pattern you're using here, I think you can just use `promises.New` instead, which might make it easier to write a slightly shorter and clearer code (as you don't need to explicitly call `callback` but just `resolve` or `reject`).

Isn't that right? @mstoykov 🤔